src/shiftIn.ts

```diff
@@ -61,7 +64,19 @@ export class ShiftInObject {
                 throw new Error("Unable to calculate gateway address");
             }
 
-            const { sendAmount, sendToken } = this.params as ShiftInFromDetails;
+            const { sendAmount, sendToken: renContract } = this.params as ShiftInFromDetails;
+
+            try {
+                // Check if the darknodes have already seen the transaction
+                const queryTxResponse = await this.queryTx();
+                if (queryTxResponse.txStatus === TxStatus.TxStatusDone || queryTxResponse.txStatus === TxStatus.TxStatusExecuting) {
```

Comment:
A pending status also indicates that the Darknodes have seen the transaction (and it is pending consensus in the transaction pool).